src/JsonLd/Util/BlankNodeIdentifiersGenerator.php

```diff
@@ -0,0 +1,49 @@
+<?php
+
+/*
+ * This file is part of the API Platform project.
+ *
+ * (c) Kévin Dunglas <dunglas@gmail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace ApiPlatform\Core\JsonLd\Util;
+
+/**
+ * Generates blank node identifiers scoped to each JSON-LD document.
+ *
+ * @author Teoh Han Hui <teohhanhui@gmail.com>
+ *
+ * @internal
+ */
+final class BlankNodeIdentifiersGenerator
+{
+    const IDENTIFIER_PREFIX = '_:b';
+
+    private $blankNodeCounts = [];
+    private $identifiers = [];
+
+    /**
+     * Gets a blank node identifier for an object, scoped to a JSON-LD document.
+     *
+     * @param object $object
+     * @param string $documentRootHash
+     *
+     * @return string
+     */
+    public function getBlankNodeIdentifier($object, string $documentRootHash): string
+    {
+        $objectHash = spl_object_hash($object);
```

Comment:
I'm not a fond of exposing such low level data from the system publicly. Is it really safe? Another potential problem is that this ID will be reused later by PHP. WDYT about using a UUID instead?